Assets/GRPC.NET/Scripts/PushPullStream.cs

```diff
@@ -35,35 +28,33 @@ public override int Read(byte[] buffer, int offset, int count)
             var readLength = 0;
             lock (m_Buffer)
             {
-                while (!ReadAvailable(count))
-                    Monitor.Wait(m_Buffer);
+                while (!ReadAvailable()) Monitor.Wait(m_Buffer);
 
-                for (; readLength < count && Length > 0 && m_Buffer.Count > 0; readLength++)
-                {
+                for (; readLength < count && m_Buffer.Count > 0; readLength++)
                     buffer[readLength] = m_Buffer.Dequeue();
```

Comment:
offset needs to be added here:
```cs
buffer[offset+readLength] = ...
```

just got bit by that bug today...

Reply:
Hi @colinrgodsey, thanks for also finding this bug. :-) I assume this happens for larger messages or how could we trigger this with a new testcase? For the initial version I tried to add a test-case for each bug I encountered.
Personally I only used high frequent short messages in production and testing until now.

Reply:
definitely a good idea keeping test cases for encountered bugs! not sure what exact internal behavior triggered this, but it did happen with a message that was larger than 128KB. i do fear the exact issue may have had to with some concurrent behavior (triggering a read on an existing buffer that hasnt been completely consumed/drained yet). although it was rather consistent in our case with this exact payload.